python/adbc_driver_manager/adbc_driver_manager/_lib.pyx

```diff
@@ -420,6 +420,22 @@ cdef class ArrowArrayStreamHandle:
         """The address of the ArrowArrayStream."""
         return <uintptr_t> &self.stream
 
+    @property
+    def is_valid(self) -> bool:
+        """Check validility (object has non-NULL release pointer)."""
+        return self.stream.release != NULL
+
+    def release(self):
+        """Release this stream without having to import it.
+
+        No-op if already released (if ``not self.is_valid``).
+
+        Postcondition: ``not self.is_valid``.
+        """
+        if self.stream.release != NULL:
+            self.stream.release(&self.stream)
+            self.stream.release = NULL
```

Comment:
is there a magic python magic func we can utilize here? like `__del__`?

Reply:
Updated

Reply:
I also added the methods to ArrowArray/ArrowSchema even though they're not used